prg_test.go

```diff
@@ -0,0 +1,100 @@
+package test
+
+import (
+	"math"
+	mrand "math/rand"
+	"testing"
+
+	. "github.com/bjartek/overflow"
+	"github.com/onflow/flow-go/crypto/random"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestNextUInt64NewPRG(t *testing.T) {
+	o, err := OverflowTesting()
+	assert.NoError(t, err)
+
+	t.Run("Should generate uniform distribution", func(t *testing.T) {
+		// make sure n is a power of 2 so that there is no bias in the last class
+		// n is a random power of 2 (from 2 to 2^10)
+		n := 1 << (1 + mrand.Intn(10))
+		classWidth := (math.MaxUint64 / uint64(n)) + 1
+		var sampleSize int
+
+		// taken from BasicDistributionTest constants - hardcoding here is
+		/// fragile, but it's determined within flow-go/crypto/rand_utils
+		if n < 80 {
+			// but if `n` is too small, we use a "high enough" sample size
+			// If n is small, use a sample size just under 85000
+			sampleSize = (85000 / n) * n // highest multiple of n less than 80000
+		} else {
+			// If n is 80 or larger, use a sample size that is n * 1000
+			sampleSize = n * 1000
+		}
+
+		// initialize PRG object in RandomResultStorage helper contract
+		seed := GetRandomBytes(t, 32)
+		salt := GetRandomBytes(t, 8)
+		InitializePRG(o, t, seed, salt)
+
+		maxBatchSize := 5000
+		// generate results in batches due to transaction computational limit
+		ProcessBatches(sampleSize, maxBatchSize, func(startIdx, batchSize int) {
+			GenerateResultsAndStore(o, t, batchSize)
+		})
+
+		// get the results in batches again due to query computational limit
+		results := make([]uint64, sampleSize)
+		ProcessBatches(sampleSize, maxBatchSize, func(startIdx, batchSize int) {
```

Comment:
apologies, my suggestion on how to structure the test wasn't simple to implement! I missed the computation limit on both the transaction (generating the randoms) and script (reading the randoms), and didn't think we may need to perform that in batches. 
Nice and clean implementation 👌🏼